### PR TITLE
Added M4_notebook.ipynb, app.py, and screenshot.png

### LINKED ISSUE #37: Step 3: Summarize from Milestone 1 to 4
`rubric={mechanics:10}`

### PR DESCRIPTION
**Pull request by rainbowxyt0305**
Hi Team,

I have draft up M4_notebook.ipynb as per our discussion, included screenshot in the img folder, and included app.py in the script folder. Please review and merge PR! Thank you!!!

### CODE REVIEW

File: notebooks/M4_notebook.ipynb [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    "* **`Milestone 3`: We setup the distributed infrastructure `EMR-Spark` cluster in cloud, and developed a Machine Learning model in `Spark` using the ready-to-use data stored in S3 from `Milestone 2`. Specifically,**\n",
+    "\n",
+    "    * we setup our `EMR` cluster with `Spark` and `TLJH`.\n",
+    "    * we setup Firefox as web browser for `EMR` and configured FoxyProxy for Firefox.\n",
+    "    * on `TLJH` running the `Spark` engine, we developed a Machine Learning model using `scikit-learn` and obtained the best hyperparameter settings using `Spark`'s `MLlib`.\n",
+    "\n",
+    "\n",
+    "* **`Milestone 4`: We deployed the Machine Learning model we trained from `Milestone 3` in cloud using `Flask` so that other consumers can use it. Specifically,**\n",
+    "\n",
+    "    * we developed our API called `app.py` using `Flask` on `TLJH`, where we created a new endpoint that accepts a POST request of the 25 features from our user and return a prediction to the user based on the trained Machine Learning model.\n",
```
Comment (tdkhanhvu) [inline notebooks/M4_notebook.ipynb]:
```suggestion
    "    * we developed our API called `app.py` using `Flask` on `TLJH`, where we created a new endpoint that accepts a POST request of the 25 features from our user and returns a prediction to the user based on the trained Machine Learning model.\n",
```

File: notebooks/M4_notebook.ipynb [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    "**Throughout these 4 milestones, our group sequentially took on the roles of 1. Data Engineer, 2. Infrastructure, 3. Data Scientist, and 4. DevOps to perform prediction tasks on the large `daily rainfall in Australia` [dataset](https://figshare.com/articles/dataset/Daily_rainfall_over_NSW_Australia/14096681) (5.7 GB), and we have achieved the following 4 objectives in each of the milestones.**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "* **`Milestone 1`: We got the data from the web using API; we processed it locally, and converted it to an efficient file format. Specifically,**\n",
+    "\n",
+    "    * we tried 3 different approaches to reduce memory usage when performing EDA in Python, including 1. changing the columns' data types in the dataframe; 2. loading only the columns we are interested in; 3. loading data in chunks.\n",
```
Comment (tdkhanhvu) [inline notebooks/M4_notebook.ipynb]: I think we can capitalize the first lettter in each bullet point

**Review by tdkhanhvu — COMMENTED** [review]
Looks good to me. Only a few modifications based on my suggestion.

File: notebooks/M4_notebook.ipynb (hunk 1 above)
Comment (rainbowxyt0305) [reply notebooks/M4_notebook.ipynb]: Thank you for the catch!

File: notebooks/M4_notebook.ipynb (hunk 2 above)
Comment (rainbowxyt0305) [reply notebooks/M4_notebook.ipynb]: Although I have `Specifically` at the end of the `Milestone x` opening sentence, I think your suggestion makes sense also - I will change it! Thank you Vu!

**Review by tdkhanhvu — APPROVED** [review]
Looks good now. Thanks Rachel

File: notebooks/M4_notebook.ipynb [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+   "metadata": {},
+   "source": [
+    "## 3. Summarize your journey from Milestone 1 to Milestone 4"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Throughout these 4 milestones, our group sequentially took on the roles of 1. Data Engineer, 2. Infrastructure, 3. Data Scientist, and 4. DevOps to perform prediction tasks on the large `daily rainfall in Australia` [dataset](https://figshare.com/articles/dataset/Daily_rainfall_over_NSW_Australia/14096681) (5.7 GB), and we have achieved the following 4 objectives in each of the milestones.**"
```
Comment (Saule-Atymtayeva) [inline notebooks/M4_notebook.ipynb:148]:
```suggestion
    "**Throughout these 4 milestones, our group sequentially took on the roles of 1. Data Engineer, 2. Infrastructure Engineer, 3. Data Scientist, and 4. DevOps to perform prediction tasks on the large `daily rainfall in Australia` [dataset](https://figshare.com/articles/dataset/Daily_rainfall_over_NSW_Australia/14096681) (5.7 GB), and we have achieved the following 4 objectives in each of the milestones.**"
```

File: notebooks/M4_notebook.ipynb [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "* **`Milestone 1`: We got the data from the web using API; we processed it locally, and converted it to an efficient file format. Specifically,**\n",
+    "\n",
+    "    * We tried 3 different approaches to reduce memory usage when performing EDA in Python, including 1. changing the columns' data types in the dataframe; 2. loading only the columns we are interested in; 3. loading data in chunks.\n",
```
Comment (Saule-Atymtayeva) [inline notebooks/M4_notebook.ipynb]:
```suggestion
    "    * We tried 3 different approaches to reduce memory usage when performing EDA in Python, including
        - 1. changing the columns' data types in the dataframe;
        - 2. loading only the columns we are interested in;
        - 3. loading data in chunks.\n",
```

Comment (Saule-Atymtayeva) [reply notebooks/M4_notebook.ipynb]: Another super minor suggestion :)

File: notebooks/M4_notebook.ipynb (hunk 3 above)
Comment (Saule-Atymtayeva) [reply notebooks/M4_notebook.ipynb:148]: Some minor suggestion to change the name of our role :)

**Review by Saule-Atymtayeva — APPROVED** [review]
Thank you for this great work, Rachel! In general, I don't have any important suggestions, just a couple of minor ones 🙂

File: notebooks/M4_notebook.ipynb (hunk 3 above)
Comment (rainbowxyt0305) [reply notebooks/M4_notebook.ipynb:148]: Thank you for the catch and great suggestion!

**Review by lephanthuymai — APPROVED** [review]
Looks great, thank you very much, Rachel!

File: notebooks/M4_notebook.ipynb (hunk 4 above)
Comment (rainbowxyt0305) [reply notebooks/M4_notebook.ipynb]: Great - I love it!

**Pull request merged**